### ISSUE TITLE
Configuration entries keep being generated for accountid = 0

### ISSUE THREAD

**Issue by laufhannes** [Runalyze/Runalyze#2042]
So far, I don't really know where they come from. This issue does only occur as long as there's no account with accountid = 0.

I only see [Category.php#L237-L239](https://github.com/Runalyze/Runalyze/blob/d8e6a74a1bbaf59abb3a3cdd7bf041c1b7091a89/inc/core/Configuration/Category.php#L237-L239) where conf entries can be generated:
```php
if (!FrontendShared::$IS_SHOWN && $this->hasUserID()) {
	$this->correctDatabaseFor($KeysInDatabase);
}
```

So, it's not possible on shared pages (or are there shared pages that do not set `FrontendShared::$IS_SHOWN` correctly?) and the user id within our (old) configuration classes has to be numeric. The configuration class uses `SessionAccountHandler::getId()` (see [Configuration.php#L111](https://github.com/Runalyze/Runalyze/blob/ac572f820228365e545ab9426dc7f667028b1708/inc/core/Configuration.php#L111)).

The [SessionAccountHandler](https://github.com/Runalyze/Runalyze/blob/master/inc/system/class.SessionAccountHandler.php#L58) uses `-1` as default (which should throw errors if used to insert data as `runalyze_conf.accountid` is unsigned) and asks [SharedLinker](https://github.com/Runalyze/Runalyze/blob/master/inc/system/class.SharedLinker.php#L95-L118) for its id, if `SharedLinker::isOnSharedPage` (which is not the same as `FrontendShared::$IS_SHOWN`). SharedLinker has a default setting of `self::USER_ID = 0`, but `getUserId()` should return -1 if no account is found, which - again - should throw an error instead of using an accountid of 0.

**Comment by laufhannes** [comment]
Pages with facebook's meta course data seem to cause that issue.

I guess `Request::param('account')` is not available (i.e. results in 0) in [class.SharedLinker.php#L106-L109](https://github.com/Runalyze/Runalyze/blob/511629f362678fc7c979b9eb0cb205650ee71320/inc/system/class.SharedLinker.php#L106-L109):
```php
	// ...
} elseif (self::isOnMetaCourseForFacebook()) {
	$Data = true;
	self::$USER_ID = (int)Request::param('account');
} else {
	// ...
```

**Comment by laufhannes** [comment]
Hopefully fixed by #2045.
